### PR TITLE
Feature/shepherd 78

### PR DESCRIPTION
**Pull request by singularo**
Enhancements to solve https://github.com/universityofadelaide/shepherd/issues/78

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -1222,8 +1222,21 @@ public function updateJob(string $name, string $image_name, array $args, array $
   /**
    * {@inheritdoc}
    */
-  public function deleteJob(string $name) {
-    return $this->apiCall(__METHOD__, $name);
+  public function deleteJob(string $name, string $label = NULL) {
```
Comment (pingers) [inline src/Client.php]: Need to update ClientInterface.

File: src/Client.php [hunk 2]
```diff
@@ -1176,8 +1176,8 @@ public function deleteCronJob(string $name, string $label = NULL) {
   /**
    * {@inheritdoc}
    */
-  public function getJob(string $name) {
-    return $this->apiCall(__METHOD__, $name);
+  public function getJob(string $name, string $label = NULL) {
```
Comment (pingers) [inline src/Client.php]: Need to update ClientInterface.

File: src/Client.php (hunk 1 above)
Comment (pingers) [inline src/Client.php]: Hmm, the code here is completely separate. I think we're better off implementing `deleteJobByLabel(string $label)` and `getJobByLabel(string $label)`.

Comment (singularo) [reply src/Client.php]: Well, there are a couple more spots that use the label thing as well, so if we're changing one, lets do them all. Do that and make this MR bigger, or do them later?

**Pull request closed without merging**